internal/discover/discover.go

```diff
@@ -86,21 +89,32 @@ func DetectServices(effxFileLocations []string) []string {
 				}
 			}
 			if !contains {
-				detectedServiceNames = append(detectedServiceNames, file.Name())
+				payload := effx_api.DetectedServicesPayload{
+					Name:       strings.ToLower(file.Name()),
+					SourceName: &sourceName,
+				}
+				if os.Getenv("DISABLE_LANGUAGE_DETECTION") != "true" {
+					result, err := metadata.InferMetadata(filepath.Dir(commonDir + file.Name()))
+					if err != nil {
+						log.Printf("Could not predict version %+v\n", err)
+					}
+
+					if result != nil && result.Language != "" && result.Version != "" {
+						payload.Tags = &map[string]string{}
+						(*payload.Tags)["language"] = strings.ToLower(result.Language)
+						(*payload.Tags)[strings.ToLower(result.Language)] = strings.ToLower(result.Version)
+					}
+				}
```

Comment:
I think you would want to break this up into two sections.
1) If you detected a language
2) If you detected both language and version

As is, you would omit language suggestions if version detection failed.

Reply:
Good call, now if version cannot be detect but language can, the language tag will be used.

Reply:
fyi, version cannot be found if a language cannot be detected.